plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.processor;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+import static org.opensearch.ml.processor.InferenceProcessorAttributes.*;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.support.GroupedActionListener;
+import org.opensearch.client.Client;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+import org.opensearch.ingest.ValueSource;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.transport.MLTaskResponse;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskAction;
+import org.opensearch.script.ScriptService;
+import org.opensearch.script.TemplateScript;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+
+/**
+ * MLInferenceIngestProcessor requires a modelId string to call model inferences
+ * maps fields in document for model input, and maps model inference output to new document fields
+ * this processor also handles dot path notation for nested object( map of array) by rewriting json path accordingly
+ */
+public class MLInferenceIngestProcessor extends AbstractProcessor implements ModelExecutor {
+
+    private final InferenceProcessorAttributes inferenceProcessorAttributes;
+    private final boolean ignoreMissing;
+    private final boolean ignoreFailure;
+    private final ScriptService scriptService;
+    private static Client client;
+    public static final String TYPE = "ml_inference";
+    public static final String DEFAULT_OUTPUT_FIELD_NAME = "inference_results";
+    public static final String IGNORE_MISSING = "ignore_missing";
```

Comment:
ignore missing for what? Let's add a comment?

Reply:
ignore_missing fields, for example, the input maps specific a field 'text', document_1 might have the field 'text' but document_2 might not have the field, then we ignore inference on  document_2  
. It's commonly seen fields in processors.
https://github.com/opensearch-project/OpenSearch/blob/207bbad6bfaffde580a8425f88627c3638547087/modules/ingest-common/src/main/java/org/opensearch/ingest/common/DissectProcessor.java#L91

Reply:
Thanks for explaining, let's add a comment as in our plugin we never used this before.

Reply:
added